capirca/lib/paloaltofw.py

```diff
@@ -466,170 +467,148 @@ def _BuildPort(self, ports):
+          member.text = fz
 
-        rules.append(self.INDENT * 9 + "<from>")
-        for fz in options["from_zone"]:
-          rules.append(self.INDENT * 10 + "<member>" + fz + "</member>")
-        rules.append(self.INDENT * 9 + "</from>")
+        from_ = ET.SubElement(entry, "from")
+        for tz in options["from_zone"]:
+          member = ET.SubElement(from_, "member")
+          member.text = tz
 
-        rules.append(self.INDENT * 9 + "<source>")
+        source = ET.SubElement(entry, "source")
         if not options["source"]:
-          rules.append(self.INDENT * 10 + "<member>any</member>")
+          member = ET.SubElement(source, "member")
+          member.text = "any"
         else:
-          for s in options["source"]:
-            rules.append(self.INDENT * 10 + "<member>" + s + "</member>")
-        rules.append(self.INDENT * 9 + "</source>")
+          for x in options["source"]:
+            member = ET.SubElement(source, "member")
+            member.text = x
 
-        rules.append(self.INDENT * 9 + "<destination>")
+        dest = ET.SubElement(entry, "destination")
         if not options["destination"]:
-          rules.append(self.INDENT * 10 + "<member>any</member>")
+          member = ET.SubElement(dest, "member")
+          member.text = "any"
         else:
-          for d in options["destination"]:
-            rules.append(self.INDENT * 10 + "<member>" + d + "</member>")
-        rules.append(self.INDENT * 9 + "</destination>")
+          for x in options["destination"]:
+            member = ET.SubElement(dest, "member")
+            member.text = x
 
-        rules.append(self.INDENT * 9 + "<service>")
+        service = ET.SubElement(entry, "service")
         if not options["service"] and not options["application"]:
-          rules.append(self.INDENT * 10 + "<member>any</member>")
+          member = ET.SubElement(service, "member")
+          member.text = "any"
         elif not options["service"] and options["application"]:
-          rules.append(self.INDENT * 10 +
-                       "<member>application-default</member>")
+          member = ET.SubElement(service, "member")
+          member.text = "application-default"
         else:
-          for s in options["service"]:
-            rules.append(self.INDENT * 10 + "<member>" + s + "</member>")
-        rules.append(self.INDENT * 9 + "</service>")
+          for x in options["service"]:
+            member = ET.SubElement(service, "member")
+            member.text = x
 
-        rules.append(self.INDENT * 9 + "<action>" + Term.ACTIONS.get(
-            str(options["action"])) + "</action>")
+        action = ET.SubElement(entry, "action")
+        action.text = Term.ACTIONS.get(options["action"])
 
+        # XXX loop vars, > 1 zones case
         if fz == tz == "any":
-          rules.append(self.INDENT * 9 + "<rule-type>interzone</rule-type>")
+          rule_type = ET.SubElement(entry, "rule-type")
```

Comment:
@kevinsteves , is `rule-type` it a valid field for 10+?

Reply:
yes, it was added in 6.1 I think.

admin@TEST-HOST5# find command keyword rule-type
set rulebase security rules <name> rule-type <universal|intrazone|interzone>
[edit] 
admin@TEST-HOST5# run show system info | match sw-version
sw-version: 10.0.4
[edit]